### PR TITLE
Add typos check to CI

### PR DESCRIPTION
**Pull request by kilpkonn**
Closes https://github.com/eclipse-zenoh/zenoh/issues/1035

### CODE REVIEW

**Comment by wyfo** [comment]
This check should also be added in `.pre-commit-config.yaml`. However, be careful when adding the hook, see https://github.com/crate-ci/typos/issues/570. I'm against letting `typos` modifying files silently at commit, so default arguments should be overridden.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -133,6 +133,16 @@ jobs:
...
+  typo-check:
+    name: Typo Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone this repository 
+        uses: actions/checkout@v4
+
+      - name: Check spelling
+        uses: crate-ci/typos@master
+
```
Comment (diogomatsubara) [inline .github/workflows/ci.yml:145]: @kilpkonn thanks for the contribution! Could you add `typo-check` to the `needs` list in the "CI status check" below?

Comment (kilpkonn) [reply .github/workflows/ci.yml:145]: Will do!

**Comment by kilpkonn** [comment]
> This check should also be added in `.pre-commit-config.yaml`. However, be careful when adding the hook, see [crate-ci/typos#570](https://github.com/crate-ci/typos/issues/570). I'm against letting `typos` modifying files silently at commit, so default arguments should be overridden.

This repo doesn't have any `pre-commit` framework stuff tho. Do you think adding it just to run `typos` is worth it/are you planning to add it anyway?

**Comment by wyfo** [comment]
> This repo doesn't have any pre-commit framework stuff tho. Do you think adding it just to run typos is worth it/are you planning to add it anyway?

Sorry, I've added a pre-commit configuration in the branch `dev/1.0.0`, not in `main`. Don't bother adding it, I will do it when we will merge `main` in `dev/1.0.0`.

**Review by diogomatsubara — APPROVED** [review]
LGTM

**Review by Mallets — APPROVED** [review]

**Pull request merged**